classes/Cart.php

```diff
@@ -3173,7 +3175,7 @@ protected function getIdCarrierFromDeliveryOption($delivery_option)
      */
     public function getDeliveryOption($default_country = null, $dontAutoSelectOptions = false, $use_cache = true)
     {
-        $cache_id = (int) (is_object($default_country) ? $default_country->id : 0) . '-' . (int) $dontAutoSelectOptions;
+        $cache_id = (int) (is_object($default_country) ? $default_country->id : 0) . '-' . (int) $dontAutoSelectOptions . '-' . (int) $this->id;
```

Comment:
I think, only this line is needed. :thinking:

Reply:
But if something is build in this order: get, set, get. The second get will not work.

Reply:
So, the cacheId should be dynamicly built and use, or maybe it's time to remove static cache :+1: